dashboard/modules/snapshot/tests/test_job_submission.py

```diff
@@ -34,10 +36,24 @@ def _get_snapshot(address: str):
 
 
 @pytest.mark.parametrize("address_suffix", ["", "/"])  # Trailing slash should succeed
+@pytest.mark.parametrize(
+    "set_override_dashboard_url",
+    [
+        None,
+        "https://external_dashboard_url",
+        "https://external_dashboard_url/path1/?query_param1=val1&query_param2=val2",
+        "new_external_dashboard_url",
+    ],
+    indirect=True,
+)
 def test_successful_job_status(
-    ray_start_with_dashboard, disable_aiohttp_cache, enable_test_module, address_suffix
+    set_override_dashboard_url,  # noqa: F811
```

Comment:
Could we actually do `monkeypatch` for the env rather than overriding sys env? Directly modifying ENV in tests created issues before AFAIK. 


For example: https://github.com/ray-project/ray/blob/6084ee5a633c4a9ac100121d24a51a40867cba1f/python/ray/tests/test_state_api.py#L1965-L1971

Reply:
Got it. I'll also update the other tests and fixtures to use monkeypatch

Reply:
thanks, i should have been more explicit in my previous nits.